resources/gradle_plugin/src/main/groovy/ConnextDdsCompileExample.groovy

```diff
@@ -0,0 +1,265 @@
+         * args.
+         * 
+         * ConfigureExample {
+         *   idlFile = "<idlFileName>"
+         *   codegenExtraArgs = [arg1, arg2, ..., argN]  
+         * }
+        */
+        def extension = project.extensions.create("ConfigureExample", ConnextDdsBuildExampleExtension)
+
+        /** Applies the java plugin */
+        project.getPluginManager().apply(JAVA_PLUGIN_NAME)
+
+        /** Sets the source dir */
+        this.setSourceDir(project, SOURCE_DIR)
+
+        /** Finds RTIConnextDDS using the env var NDDSHOME */
+        def connextDdsPath = new File(System.getenv("NDDSHOME"))
+        def libsPath = new File(connextDdsPath, "lib/java")
+
+        /** Adds the directory to find the RTIConnextDDS Java libraries */
+        project.getRepositories().flatDir {
+            dirs libsPath
+        }
+
+        /** Adds the dependency at compile and run time */
+        project.getDependencies().add("implementation", ":nddsjava")
+
+        /** Adds and configures the created tasks */
+        this.addCodegenTask(project, extension)
+        this.addDeleteGeneratedFilesTask(project, extension)
+        this.addAndConfigureApplicationPlugin(project, extension)
+    }
+
+    /**@
+     * Sets the source directory with the path which contains the .java files.
+     *
+     * @param project the gradle's project object
+     * @param sourceDir the source directory to set
+     */
+    private void setSourceDir(Project project, String sourceDir)
+    {
+        def sourceSets = project
+            .getExtensions()
+            .getByType(SourceSetContainer.class)
+        def main = sourceSets.getByName(SourceSet.MAIN_SOURCE_SET_NAME)
+        main.getJava().srcDir(sourceDir)
+    }
+
+    /**@
+     * Adds the codegen task to the project. This task calls rtiddsgen to
+     * generate all the required source files for building the examples.
+     *
+     * @param project the gradle's project object
+     * @param extension contains the name of the idlFile and the codegen extra
+     * args
+     */
+    private void addCodegenTask(
+            Project project,
+            ConnextDdsBuildExampleExtension extension)
+    {
+        /** Adds the codegen task */
+        def codegen = project.task(CODEGEN_TASK_NAME) {
+            doFirst {
+                /** There are a few examples without .idl file */
+                if(extension.idlFile.get()) {
+                    def codegenExecutableName = "rtiddsgen"
+
+                    /** 
+                     * -ppDisable is required for compiling in Windows without
+                     * needing Visual Studio or Ninja. In the future it would
+                     * be convenient to delete this argument and create a new
+                     * task to find Visual Studio or Ninja.
+                     */
+                    def codegenArgs = [
+                        "-language",
+                        "Java",
+                        "-ppDisable",
```

Comment:
why do we need to disable the preprocessor? Probably with the IDLs that we currently have it is not needed, but if somebody adds a new example that requires it, there is no way of enabling it without modifying this configuration.

Reply:
We talked about it. It is disabled because it was causing problems in Windows but I can remove this argument.

Reply:
But today it is not a problem so I will remove it.